src/main/groovy/io/fourfinanceit/pipeline/example/jobs/BuildAndPublish.groovy

```diff
@@ -22,29 +29,22 @@ import static io.fourfinanceit.pipeline.example.JenkinsVariable.SCM_POLLING_ENAB
  * @author Marek Kapowicki
  */
 class BuildAndPublish extends MicroserviceJobDefinition {
+    
     @Override
     void configure(Job job, MicroserviceProject project, JenkinsVariables jenkinsVariables) {
-        job.with {
-            setupEnvironmentVariables(job)
-            configure JobDslConfigureHacksUtil.injectAndMaskPasswords()
-            scm MicroserviceJobUtils.configureGit(project)
-            if (project.scmConfig.cronToPoll && jenkinsVariables.getBoolean(SCM_POLLING_ENABLED, true)) {
-                triggers {
-                    scm(project.scmConfig.cronToPoll)
-                }
-            }
-            addGradleBuildStep(job, project)
-            //addArchiveJunitAsBuildStep(job)
-            steps {
-                ifBuildStillSuccessful(delegate as StepContext) {
-                    gradle("publish --stacktrace ${projectVersionParameters()}")
-                }
-            }
-        }
+        (
+            setupEnvironmentVariables() &
+                maskPasswords() &
+                configureGit(project) &
+                use(onCron(project.scmConfig.cronToPoll))
+                    .when(scmPoolingEnabled(jenkinsVariables)) &
+                buildProject() &
+                publishArtifacts()
+        ).applyOn(job)
```

Comment:
if these parentheses are indeed needed, the JobComponents has a serious flaw of requiring it - or having not-the-job-we're-thinking-about. I don't think this can be fixed by choosing a different operator, as '.' is one of the strongest-binding. What we could do though, is making applyOn private, and having a static `JobComponent.applyOn(job, jobComponent)` method. That would force a variable extraction for the jobComponent, or at least would structure the expression naturally, without fear of applying it too eagerly.